core/src/main/java/org/apache/calcite/sql/SqlDeclareHandler.java

```diff
@@ -86,6 +90,21 @@ public SqlDeclareHandler(SqlParserPos pos, HandlerType handlerType,
     }
   }
 
+  @Override public void validate(SqlValidator validator,
```

Comment:
Sorry I'm a little slow on the uptake but what does the validate function do? It doesn't seem to raise an error if the input isn't valid and it doesn't return anything either.

Reply:
This validate function is used to find SqlConditionDeclaration references from the BlockScope and adds it to the class field `conditionDeclarations`.

Reply:
Sure sounds good